src/codeFly/dispatcher/README.md

```diff
@@ -0,0 +1,106 @@
+# Interfaces  
+Here are basic interfaces for front-end team and back-end team to follow:
+
+## Register
+The user is required to input ```username```, ```password```, and ```email address``` on the GUI. Once the user clicks the ```Register``` button, a post request will be sent to the server. Then the server will send a response to indicate if this registration is successful or not in the content body. The user will be navigated to different web pages based on the response.
+```
+Request(font-end)  
+Request Method: POST  
+URI: localhost:8080/register  
+Request Body:  
+Username: xxxxx  
+Password: xxxxx  
+Email: xxx@gmail.com  
+
+Response(back-end)
+Response Body:
+Success/Fail
+```
+
+## Login
+The user is supposed to input ```username``` and ```password``` to login. After the user clicks the ```Log In``` button, a post request will be sent to the server. Then the server will send a response to indicate if this login is successful or not in the content body. The user will be navigated to different web pages based on the response.
+```
+Request(font-end)  
+Request Method: POST  
+URI: localhost:8080/login  
+Request Body:  
+Username: xxxxx   
+Password: xxxxx  
+
+Response(back-end)  
+Ressponse Body:  
+Success/Fail  
+```
+
+## Question List
+Users who successfully login will be navigated to the main control view of their own account. There is supposed to be a ```Get Question List``` button. Once clicked, a get request will be sent to the server to ask for all the questions’ number and their titles. The server will send a response which contains a list of ```key/value``` pair in the content body to indicate all the question numbers and corresponding titles.
+```
+Request(font-end  )
+Request Method: GET  
+URI: localhost:8080/questions  
+Request Body:  
+
+Response(back-end)  
+Response Body:  
+Question1: Two Sum  
+Question2: Reverse Linked List   
+…  
+```
+
+## Start
+The user is supposed to be navigated to the view of question list after clicking the ```Get Question List``` button. Among all the questions, the user can select one and click the ```Start``` button to start programming. A post request will be sent to the server. Then the server will send a response which contains the question description in the content body to the user.
+```
+Request(font-end)  
+Request Method: POST  

```

Comment:
`GET` with a query string in the URI should suffice. For example:
`GET /start/user=jhp&qnum=1 HTTP/1.1`